packages/panzer/disc-fe/src/evaluators/Panzer_Integrator_GradBasisCrossVector_impl.hpp

```diff
@@ -137,18 +137,20 @@ namespace panzer
 
     // Create the fields that we're either contributing to or evaluating
     // (storing).
-    fields_.clear();
-    for (const auto& name : resNames)
+    fields_ = Kokkos::View<PHX::MDField<ScalarT, Cell, BASIS>*>("Integrator_GradBasisCrossVector::fields_", resNames.size());
+
     {
-      MDField<ScalarT, Cell, BASIS> res(name, basis.functional);
-      fields_.push_back(res);
+      int i=0;
+      for (const auto& name : resNames)
+        fields_(i++) = MDField<ScalarT, Cell, BASIS>(name, basis.functional);
     } // end loop over resNames
-    for (const auto& field : fields_)
+    for (std::size_t i=0; i< fields_.extent(0); ++i) {
+      const auto& field = fields_(i);
       if (evalStyle_ == EvaluatorStyle::CONTRIBUTES)
         this->addContributedField(field);
       else // if (evalStyle_ == EvaluatorStyle::EVALUATES)
         this->addEvaluatedField(field);
-
+    }
```

Comment:
I think you probably want the `if` outside the `for`.

Reply:
Do you want to switch this up such that we're not doing an `if` every iteration?  If not, I'll dismiss my review and you can merge.